kernel/aster-nix/src/process/posix_thread/futex.rs

```diff
@@ -423,17 +431,31 @@ impl FutexWaiter {
         }
     }
 
-    pub fn wait(&self) {
+    pub fn wait(&self, timeout: Option<FutexTimeout>) {
         let current_thread = current_thread!();
         if current_thread.tid() != self.tid {
             return;
         }
         self.is_woken.store(false, Ordering::SeqCst);
-        while !self.is_woken() {
-            // debug!("futex is wait for waken, tid = {}", self.tid);
-            Thread::yield_now();
+
+        if let Some(timeout) = timeout {
+            let wait_queue = WaitQueue::new();
```

Comment:
The `wait_queue` is created as a local variable, so the `wake` method of this `wait_queue` can never be called. 

I suggest putting the `wait_queue` in `FutexWaiter`, and the method `FutexWaiter::wake` should also wake up the `wait_queue`